.pylintrc

```diff
@@ -454,14 +470,32 @@ disable=raw-checker-failed,
         file-ignored,
         suppressed-message,
         useless-suppression,
-        deprecated-pragma,
+        deprecated-pragma, # because we have some peniding deprecations in the code. 
         use-symbolic-message-instead,
         use-implicit-booleaness-not-comparison-to-string,
         use-implicit-booleaness-not-comparison-to-zero,
-        no-else-return,
+        no-else-return, # this is a style preference, we don't need to follow it
         inconsistent-return-statements,
-        unspecified-encoding,
+        unspecified-encoding, # this is not a relevant issue.
         no-member, # because we use funcify_method decorator
+        invalid-overridden-method, # because we use funcify_method decorator
+        too-many-function-args, # because we use funcify_method decorator
+        fixme, # because we use TODO in the code
+        missing-module-docstring, # not useful for most of the modules.
+        attribute-defined-outside-init, # to avoid more than 200 errors (code works fine)
```

Comment:
> Oh man this was one of the problems that I was excited to see you fix
> 
> So we are just giving up on this? It seems like bad practice to have a class that if you call an attribute from an object it will break at some points, and work at others.

I can't tell you that we are giving up on this forever.  But I ensure to you that I already gave up on this a few weeks ago... It is **extremely** hard to fix more than 200 pylint errors quickly. Please take in mind how hard it was to run pylint on more than 10k code lines, I had to simplify some of the points.

Also, there is something worser that you didn't mention...

> Then again, we generally avoid most of these cases and very rarely had any problems with them. But still, I can't be sure we avoid all of them. This check would fix any potential issues...

... The only issue I can see is to an user trying to access an atribute that was not set in the simulation yet. But in that case, what are you going to return to the user if the variable didn't have their value calculated yet?

For example, imagine we define self.x (x is of Function type) outside of the constrictor (init method). So, to satisfy pylint, we set self.x = None in the init method. If the users expects a Function type but the code returned - without errors - the None type, are we really making a better service to the final user?

Additionally, do you have any other source declaring this as a "bad practice" rather than just pylint? I feel like pylint created their own rules based on `nothing`. I can't find where in PEP8, for example, they recommend to always define a  variable within the init method; 

Make no mistake, I do like the idea of defining all the class atributes in the init method. But it is a quite consuming taks and I'm not totally convinced about it yet.